ksql-parser/src/main/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercer.java

```diff
@@ -15,13 +15,19 @@
 
```

Comment:
Now that I'm looking at the history, I remember that it was originally like that and then @big-andy-coates refactored it to extract out a coerce that could be used general purpose in https://github.com/confluentinc/ksql/pull/2953. I'm usually a fan of making things general purpose only if they are needed to be, but I'll let Andy chime in before making that change.

Reply:
At the moment we have code that coerces values all over the shot, (e.g. the UDF coercion code that Tim recently removed, but also others in the serde code and some more somewhere that coerces GenericRow),  this class is only _currently_ used in one place,  but the intent is that we slowly move the other areas to use one common class to handle coercion of our SQL types, so that we get consistent behaviour.

Reply:
That said, this class currently does two things: 

1. It handles our implicit conversion from STRING to DECIMAL - which I really don't like. I hate such magic.
1. It handles our implicit casting rules.

Personally, I'd like to remove the first thing from our system entirely.  We have a major release coming up imminently, which gives us the opportunity to make a breaking change and handle decimals as first class citizens. Personally, I think the literal `12.453835` in a statement should be a decimal, (as per our `SqlBase` rules), but its currently a double.  We can choose to change this in the release after 5.4.

Food for thought.